analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java

```diff
@@ -0,0 +1,743 @@
+		JUnitUtil.setUpRepositoryMock(mockActivityRepository);
+		JUnitUtil.setUpRepositoryMock(mockDayActivityRepository);
+	}
+
+	private Map<Locale, String> usString(String string)
+	{
+		return Collections.singletonMap(Translator.EN_US_LOCALE, string);
+	}
+
+	private Set<ActivityCategoryDto> getAllActivityCategories()
+	{
+		return goalMap.values().stream().map(goal -> ActivityCategoryDto.createInstance(goal.getActivityCategory()))
+				.collect(Collectors.toSet());
+	}
+
+	@Test
+	public void updateLastMonitoredActivityDateIfRelevant_default_callsUserAnonymizedEntityHolderGetEntity()
+	{
+		UserAnonymizedEntityHolder mockUserAnonymizedEntityHolder = Mockito.mock(UserAnonymizedEntityHolder.class);
+		when(mockUserAnonymizedEntityHolder.getEntity()).thenReturn(userAnonEntity);
+		ZonedDateTime t1 = now();
+
+		service.updateLastMonitoredActivityDateIfRelevant(mockUserAnonymizedEntityHolder, createPayload(t1, t1));
+
+		verify(mockUserAnonymizedEntityHolder, atLeastOnce()).getEntity();
+	}
+
+	@Test
+	public void addActivity_default_callsUserAnonymizedEntityHolderGetEntity()
+	{
+		UserAnonymizedEntityHolder mockUserAnonymizedEntityHolder = Mockito.mock(UserAnonymizedEntityHolder.class);
+		when(mockUserAnonymizedEntityHolder.getEntity()).thenReturn(userAnonEntity);
+		ZonedDateTime t1 = now();
+
+		service.addActivity(mockUserAnonymizedEntityHolder, createPayload(t1, t1), GoalDto.createInstance(gamblingGoal),
+				Optional.empty());
+
+		verify(mockUserAnonymizedEntityHolder, atLeastOnce()).getEntity();
+	}
+
+	@Test
+	public void addActivity_noLastRegisteredActivity_goalConflictMessageCreated()
+	{
+		UserAnonymizedEntityHolder userAnonymizedEntityHolder = new UserAnonymizedEntityHolder(mockUserAnonymizedService,
+				userAnonId);
+		ZonedDateTime t1 = now();
+
+		service.addActivity(userAnonymizedEntityHolder, createPayload(t1, t1), GoalDto.createInstance(gamblingGoal),
+				Optional.empty());
+
+		verifyGoalConflictMessageCreated(gamblingGoal);
+	}
+
+	@Test
+	public void addActivity_timeZoneGoal_noGoalConflictMessagesCreated()
+	{
+		UserAnonymizedEntityHolder userAnonymizedEntityHolder = new UserAnonymizedEntityHolder(mockUserAnonymizedService,
+				userAnonId);
+		ZonedDateTime t1 = now();
+
+		service.addActivity(userAnonymizedEntityHolder, createPayload(t1, t1), GoalDto.createInstance(shoppingGoal),
+				Optional.empty());
+
+		verifyNoGoalConflictMessagesCreated();
+	}
+
+	@Test
+	public void addActivity_nonZeroBudgetGoal_noGoalConflictMessagesCreated()
+	{
+		UserAnonymizedEntityHolder userAnonymizedEntityHolder = new UserAnonymizedEntityHolder(mockUserAnonymizedService,
+				userAnonId);
+		ZonedDateTime t1 = now();
+
+		service.addActivity(userAnonymizedEntityHolder, createPayload(t1, t1), GoalDto.createInstance(socialGoal),
+				Optional.empty());
+
+		verifyNoGoalConflictMessagesCreated();
+	}
+
+	@Test
+	public void addActivity_afterConflictIntervalLastRegisteredActivity_newGoalConflictMessageCreated()
```

Comment:
I wouldn't add 'new' in the name. Anything you create is new, right? :smiley:

Reply:
OK, will change